core/genesis.go

```diff
@@ -155,15 +158,24 @@ func hashAlloc(ga *types.GenesisAlloc, isVerkle bool) (common.Hash, error) {
 			statedb.SetState(addr, key, value)
 		}
 	}
-	return statedb.Commit(0, false)
+
+	stateRoot, err := statedb.Commit(0, false)
+	if err != nil {
+		return common.Hash{}, common.Hash{}, err
+	}
+	// get the storage root of the L2ToL1MessagePasser contract
+	storageRootMessagePasser := statedb.GetStorageRoot(params.OptimismL2ToL1MessagePasser)
```

Comment:
This storage lookup should be conditional on isthmus. Although it may return a zeroed hash, we don't want code-paths to run if they don't need to run, to derisk changes related to a hardfork before/outside said hardfork.

Reply:
Done.